coalib/settings/Setting.py

```diff
@@ -216,6 +219,7 @@ def __path__(self, origin=None, glob_escape_origin=False):
         :raises ValueError:        If no origin is specified in the setting
                                    nor the given origin parameter.
         """
+        origin = origin.file
```

Comment:
this doesnt belong here.  it wont be backwards compatible here.

See below, which has `if hasattr(self, 'origin') ...`

To be backwards compatible you will need to add `if isinstance(origin, SourcePosition) ..`